Python/ceval_gil.h

```diff
@@ -243,6 +283,18 @@ take_gil(struct _ceval_runtime_state *ceval, PyThreadState *tstate)
 
     MUTEX_UNLOCK(gil->mutex);
 
+    if (thread_must_exit(tstate)) {
```

Comment:
This code path is to prevent https://bugs.python.org/issue39877#msg363667 crash. Are you suggesting to move this code at line 249, after COND_TIMED_WAIT()?

thread_must_exit() at entry is to prevent https://bugs.python.org/issue39877#msg363512 crash.